common/src/main/java/net/william278/huskchat/message/PrivateMessage.java

```diff
@@ -49,6 +50,11 @@ public void dispatch() {
                         continue;
                     }
                     if (!sender.hasPermission("huskchat.command.socialspy")) {
+                        try {
+                            PlayerCache.removeSocialSpy(sender);
+                        } catch(IOException e) {
+                            implementor.getLoggingAdapter().log(Level.SEVERE, "Failed to remove social spy after failed permission check");
```

Comment:
Small snag but don't forget to log the error so it's properly printed, too.
`implementor.getLoggingAdapter().log(Level.SEVERE, "Failed to remove social spy after failed permission check", e);`